src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableExtensions.cs

```diff
@@ -19,7 +19,7 @@ internal static bool IsValueType<T>()
 #if NETCOREAPP
             return typeof(T).IsValueType;
 #else
-            if (default(T) != null)
+            if (typeof(T).IsValueType)
```

Comment:
This is a downlevel path.  This may be used in runtimes where typeof(T).IsValueType isn't an intrinsic and isn't optimized.  Note that this is part of an `#else` where the `#if NETCOREAPP` already uses IsValueType.